src/ConnectionRequest.jl

```diff
@@ -23,18 +23,47 @@ abstract type ConnectionPoolLayer{Next <: Layer} <: Layer end
 export ConnectionPoolLayer
 
 function request(::Type{ConnectionPoolLayer{Next}}, url::URI, req, body;
+                 proxy=nothing,
                  socket_type::Type=TCPSocket, kw...) where Next
 
+    if proxy != nothing
+        target_url = url
+        url = URI(proxy)
+        if target_url.scheme == "http"
+            req.target = string(target_url)
+        end
+    end
+
     IOType = ConnectionPool.Transaction{sockettype(url, socket_type)}
     io = getconnection(IOType, url.host, url.port; kw...)
 
     try
+
+        if proxy != nothing && target_url.scheme == "https"
+            target = "$(URIs.hoststring(target_url.host)):$(target_url.port)"
+            @debug 1 "📡  CONNECT HTTPS tunnel to $target"
+            r = Request("CONNECT", target)
+            writeheaders(io, r)
+            startread(io)
+            readheaders(io, r.response)
+            if r.response.status != 200
+                return r
+            end
+            io = ConnectionPool.sslupgrade(io, target_url.host; kw...)
+        end
+
         return request(Next, io, req, body; kw...)
+
     catch e
         @debug 1 "❗️  ConnectionLayer $e. Closing: $io"
         close(io)
         rethrow(isioerror(e) ? IOError(e, "during request($url)") : e)
+    finally
+        if proxy != nothing && target_url.scheme == "https"
+            close(io)
+        end
```

Comment:
Avoid reusing the connection to the proxy because it got upgraded to be a HTTPS tunnel, so we can't use it to talk to the proxy anymore.